File: ts/Home/backward_each_word.ts
#!/usr/bin/env checkio --domain=js run backward-each-word

// In a given string you should reverse every word, but the words should stay in their places.
// 
// Input:A string.
// 
// Output:A string.
// 
// Precondition:The line consists only from alphabetical symbols and spaces.
// 
// 
// END_DESC

// const assert = require('assert');
// function backwardStringByWord(text) {

import assert from "assert";
function backwardStringByWord(text: string): string {
    return text.split(' ').map((str) => str.split('').reverse().join('')).join(' ');
}

console.log('Example:');
console.log(backwardStringByWord('hello    world'));

// These "asserts" are used for self-checking
assert.equal(backwardStringByWord(''), '');
assert.equal(backwardStringByWord('world'), 'dlrow');
assert.equal(backwardStringByWord('hello world'), 'olleh dlrow');
assert.equal(backwardStringByWord('hello   world'), 'olleh   dlrow');
assert.equal(backwardStringByWord('welcome to a game'), 'emoclew ot a emag');

console.log("Coding complete? Click 'Check' to earn cool rewards!");

File: py_checkio_solutions/Home/backward_each_word.py
#!/usr/bin/env checkio --domain=py run backward-each-word

# In a given string you should reverse every word, but the words should stay in their places.
# 
# Input:A string.
# 
# Output:A string.
# 
# Precondition:The line consists only from alphabetical symbols and spaces.
# 
# 
# END_DESC

def backward_string_by_word(text: str) -> str:

    # Test string
    # text = "Hello How Are You"

    # Special case 1, Empty String
    if text == '':
        return ''
   
    li = text.split(' ')
    ans = []

    for i in range(0, len(li)):
        old_word = li[i]
        ans.append(old_word[::-1])
        # if old_word != '':
        #     new_word = old_word[::-1]
        #     #print(new_word)
        #     ans.append(new_word)
        # else:
        #     ans.append('')
    
    return ' '.join(ans)

if __name__ == '__main__':
    print("Example:")
    print(backward_string_by_word(''))

    # These "asserts" are used for self-checking and not for an auto-testing
    assert backward_string_by_word('') == ''
    assert backward_string_by_word('world') == 'dlrow'
    assert backward_string_by_word('hello world') == 'olleh dlrow'
    assert backward_string_by_word('hello   world') == 'olleh   dlrow'
    assert backward_string_by_word('welcome to a game') == 'emoclew ot a emag'
    print("Coding complete? Click 'Check' to earn cool rewards!")